internal/stream/stream_test.go

```diff
@@ -250,3 +251,150 @@ func TestPushStream_Stop(t *testing.T) {
 	err := ps.Stop()
 	assert.Nil(t, err)
 }
+
+func TestNewPushStream_Failure(t *testing.T) {
+	ctx := context.Background()
+	ctrl := gomock.NewController(t)
+	subscriptionCoreMock := mocks1.NewMockICore(ctrl)
+	subscriberCoreMock := mocks2.NewMockICore(ctrl)
+	workerID := uuid.New().String()
+	httpConfig := &httpclient.Config{}
+	type args struct {
+		ctx              context.Context
+		nodeID           string
+		subName          string
+		subscriptionCore subscription.ICore
+		subscriberCore   subscriber.ICore
+		config           *httpclient.Config
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *PushStream
+		wantErr bool
+	}{
+		{
+			name: "New Push Stream Failure",
+			args: args{
+				ctx:              ctx,
+				nodeID:           workerID,
+				subName:          subName,
+				subscriptionCore: subscriptionCoreMock,
+				subscriberCore:   subscriberCoreMock,
+				config:           httpConfig,
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			subscriptionCoreMock.EXPECT().Get(gomock.Any(), subName).DoAndReturn(
+				func(arg0 context.Context, arg1 string) (*subscription.Model, error) {
+					if tt.wantErr {
+						return nil, errors.New("Test Error")
+					}
+					return getMockSubModel(""), nil
+				},
+			)
+			got, err := NewPushStream(tt.args.ctx, tt.args.nodeID, tt.args.subName, tt.args.subscriptionCore, tt.args.subscriberCore, tt.args.config)
+			if err == nil {
+				assert.Equalf(t, tt.want, got, "NewPushStream(%v, %v, %v, %v, %v, %v)", tt.args.ctx, tt.args.nodeID, tt.args.subName, tt.args.subscriptionCore, tt.args.subscriberCore, tt.args.config)
+			}
+			assert.Equal(t, got, tt.want)
+			assert.Equal(t, tt.wantErr, err != nil)
+		})
+	}
+}
+
+func TestStart_RestartSubscriber(t *testing.T) {
+	ctx := context.Background()
+	ctrl := gomock.NewController(t)
+
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(`{"IsSuccess": true}`))
+	}))
+
+	defer s.Close()
+
+	tests := []struct {
+		name      string
+		endpoint  string
+		isSuccess bool
+	}{
+		{
+			name:      "Run Push Stream Restart Subscriber",
+			endpoint:  s.URL,
+			isSuccess: true,
+		},
+	}
+
+	for _, test := range tests {
+		ps, sub := getMockPushStreamRestart(ctx, ctrl, test.endpoint)
```

Comment:
why defined a separate mock push stream here, and not used getMockPushStream() instead..?

Reply:
In this test case to restart subscriber I need to multiple changes to the existing one getMockPushStream:
1. I need to return subscriber from the getMockPushStream to compare the old and new subscriber. This needed changes in the existing code.
2. I need to mock getErrorChannel in the getMockSubscriber
3. I need to change the times  of NewSubscriber for restart subscriber
That's why chose to make a new mockPushStream

Reply:
@sumitkrzp Lets try to reuse the same mock subscriber by making some changes to the existing mock if possible.

Reply:
I tried changing the existing one but i have to make multiple changes since mockSubscriber is written for one time expect and in restart Subscriber we mock to get expect for 2 times, there will be too much if else conditions. So i am using new one only

Reply:
You can just write AnyTimes for the expected func calls. No need to make it condiitonal

Reply:
There was mistake in the test case i was compairing the subscribercoremock with mocksubcriber. fixed the issue. func signature is different now new method is needed.